src/BackendRequest.cpp

```diff
@@ -61,7 +62,49 @@ namespace NextPVR
     tinyxml2::XMLDocument doc;
     return DoMethodRequest(resource, doc, false) == tinyxml2::XML_SUCCESS;
   }
+  /*
+  tinyxml2::XMLError Request::DoCachedRequest(std::string resource, tinyxml2::XMLDocument& doc, std::string cacheFile)
+  {
+    const std::string filename = kodi::tools::StringUtils::Format("%s%s.cache", m_settings->m_instanceDirectory.c_str(), resource.c_str());
+    gzFile gz_file;
+    //open the file for writing in binary mode
+    gz_file = gzopen(kodi::vfs::TranslateSpecialProtocol(filename).c_str(), "wb");
+
+    std::stringstream data;
+    //Fill stream with data
+
+    //Get the size of the stream
+    unsigned long int file_size = sizeof(char) * response.size();
+    //Write the size of the stream, this is needed so that we know
+    //how much to read back in later
+    gzwrite(gz_file, (void*)&file_size, sizeof(file_size));
+    //Write the data
+    gzwrite(gz_file, (void*)(response.c_str()), file_size);
+    //close the file
+    gzclose(gz_file);
 
+    //open the file for reading in binary mode
+    gz_file = gzopen("c:/temp/filename.dat", "rb");
+    //this variable will hold the size of the file
+    unsigned long int size;
+    //we wrote out a unsigned long int when storing the file
+    //read this back in to get the size of the uncompressed data
+    gzread(gz_file, (void*)&size, sizeof(size));
```

Comment:
This looks like test code. Is it by intention to have it here?

Reply:
Certainly was